packages/core/src/app/formFields/getCustomFormFieldsValidationSchema.ts

```diff
@@ -62,7 +61,6 @@ export default memoize(function getCustomFormFieldsValidationSchema({
                         schema[name] = number()
                             // Transform NaN values to undefined to avoid empty string (empty input) to fail number
                             // validation when it's optional
-                            .strict(true)
```

Comment:
The `strict` option should never be used with `transform`, as it is designed to bypass the `transform` function. The reason why it worked before the version upgrade was due to a bug in the library. The bug has been fixed since version [0.29.1](https://github.com/jquense/yup/blob/master/CHANGELOG.md#bug-fixes-31).